modules/layers/src/text-layer/font-atlas-utils.js

```diff
@@ -0,0 +1,66 @@
+// TODO merge with icon-layer/icon-manager
+export function nextPowOfTwo(number) {
+  return Math.pow(2, Math.ceil(Math.log2(number)));
+}
+
+/**
+ * Generate character mapping table or update from an existing mapping table
+ * @param characterSet {Array|Set} new characters
+ * @param getFontWidth {Function} function to get width of each character
+ * @param fontHeight {Number} height of font
+ * @param buffer {Number} buffer surround each character
+ * @param maxCanvasWidth {Number} max width of font atlas
+ * @param mapping {Object} old mapping table
+ * @param xOffset {Number} x position of last character in old mapping table
+ * @param yOffset {Number} y position of last character in old mapping table
+ * @returns {{
+ *   mapping: Object,
+ *   xOffset: Number, x position of last character
+ *   yOffset: Number, y position of last character in old mapping table
+ *   canvasHeight: Number, height of the font atlas canvas, power of 2
+ *  }}
+ */
+export function buildMapping({
+  characterSet,
+  getFontWidth,
+  fontHeight,
+  buffer,
+  maxCanvasWidth,
+  mapping = {},
+  xOffset = 0,
+  yOffset = 0
+}) {
+  let row = 0;
+  // continue from x position of last character in the old mapping
+  let x = xOffset;
```

Comment:
This `buildMapping` changed to allow appending new characters to existing mapping

Old code
```
  const mapping = {};
  let row = 0;
  let x = 0;
```